Sources/TripKitUI/helper/RxTripKit/TKServer+Rx.swift

```diff
@@ -75,7 +75,12 @@ extension Reactive where Base: TKServer {
     Single.create { subscriber in
       base.hit(method, path: path, parameters: parameters, headers: headers, region: region
       ) { status, headers, result in
-        subscriber(.success((status, headers, try? result.get())))
```

Comment:
@nighthawk I'm not sure why we silenced the error here. I have caller like this,

```
TGAccount.requireUserToken()
      .flatMap { _ in TKServer.shared.rx.hit(path: "data/user") }
      .subscribe(onSuccess: { (status, _, _) in
        // The user token is valid, nothing to do.
        assert(status == 200, "unexpected status code")
      }, onFailure: { (error) in
        let code = (error as NSError).code
        if code == 447 {
          // See: https://redmine.buzzhives.com/issues/13930 for why
          // why catch this error code
          TGAccount.update(asSignedIn: false, viaExternal: true)
        } else {
          TKLog.error("Auth0ViewModel", text: "Error is \(error)")
        }
      })
      .disposed(by: disposeBag)
```

By silencing the error, the `success` block always gets executed. I think that is fine, if I remove the error handling code (by checking the status code) to the `success` block. But contextually it seems less appropriate. 

Maybe there was a reason for silencing the error, in that case, please advise and I'll make the call handle differently.

Reply:
There's a reason for this, which is that some endpoints don't return any data (just a status) but this then throws a `ServerError.noData`, and then all uses of those endpoints would come back as an `.error` case and all those would fail.

We could address this two ways:

- Catch the `ServerError.noData` error here and only in that case return a .success with `nil` data, otherwise throw
- Add a parameter which says whether this should fail or not, if there was no data returned
- Add a dedicated method for the endpoints that return no data

I think the first option is the best.